Add Fleet & Agent 8.16.0 Release Notes (#1412)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.16.asciidoc

:fleet-server-issue: https://github.com/elastic/fleet-server/issues/
:fleet-server-pull: https://github.com/elastic/fleet-server/pull/

[[release-notes]]
= Release notes

This section summarizes the changes in each release.

* <<release-notes-8.16.0>>

Also see:

* {kibana-ref}/release-notes.html[{kib} release notes]
* {beats-ref}/release-notes.html[{beats} release notes]

// begin 8.16.0 relnotes

[[release-notes-8.16.0]]
== {fleet} and {agent} 8.16.0

Review important information about the {fleet} and {agent} 8.16.0 release.

[discrete]
[[security-updates-8.16.0]]
=== Security updates

{fleet-server}::
* Update {fleet-server} Go version to 1.23.1. {fleet-server-pull}3924[#3924]

[discrete]
[[new-features-8.16.0]]
=== New features

The 8.16.0 release Added the following new and notable features.

{fleet}::
* Support content packages in UI. {kibana-pull}195831[#195831]
* Advanced agent monitoring options UI for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]
* Add option to have Kafka dynamic topics in outputs. {kibana-pull}192720[#192720]
* Add support for GeoIP processor databases in Ingest Pipelines. {kibana-pull}190830[#190830]

> * Add agentless UX creation flow. {kibana-pull}189932[#189932]

Reviewer: @kpollich do we want anything to do with agentless in the release notes yet, considering it isn't usable or really announced yet?

* Enable feature flag for reusable integration policies. {kibana-pull}187153[#187153]

{fleet-server}::

> * Add `/api/fleet/agents/:id/audit/unenroll` API that an {agent} or Endpoint process may use to enable accurate status reporting in the {fleet} UI. {fleet-server-pull}3818[#3818] {agent-issue}484[#484]

Reviewer: ```suggestion
* Add `/api/fleet/agents/:id/audit/unenroll` API that an {agent} or Endpoint process may use to report that an agent was uninstalled or unenrolled to {fleet}. {fleet-server-pull}3818[#3818] {agent-issue}484[#484]
```

* Add a `secret_paths` attribute to the policy data sent to agents. This attribute is a list of keys that {fleet-server} has replaced with a reference to a secret value. {fleet-server-pull}3908[#3908] {fleet-server-issue}3657[#3657] 

{agent}::
* Uninstalling a {fleet}-managed {agent} instance will now do a best-effort attempt to notify {fleet-server} of the agent removal so the agent status appears correctly in the {fleet} UI (related to {fleet-server-pull}3818[#3818] above). {agent-pull}5302[#5302] {agent-issue}484[#484]

> * Remove support for the experimental shippers feature. Removal is non-breaking since the feature was experimental only. {agent-pull}5308[#5308] {agent-issue}4547[#4547]

Reviewer: ```suggestion
* Remove support for the experimental shippers feature. {agent-pull}5308[#5308] {agent-issue}4547[#4547]
```

:beats-issue: https://github.com/elastic/beats/issues/
:beats-pull: https://github.com/elastic/beats/pull/
:agent-libs-pull: https://github.com/elastic/elastic-agent-libs/pull/
:agent-issue: https://github.com/elastic/elastic-agent/issues/
:agent-pull: https://github.com/elastic/elastic-agent/pull/
:fleet-server-issue: https://github.com/elastic/fleet-server/issues/
:fleet-server-pull: https://github.com/elastic/fleet-server/pull/

[[release-notes]]
= Release notes

This section summarizes the changes in each release.

* <<release-notes-8.16.0>>

Also see:

* {kibana-ref}/release-notes.html[{kib} release notes]
* {beats-ref}/release-notes.html[{beats} release notes]

// begin 8.16.0 relnotes

[[release-notes-8.16.0]]
== {fleet} and {agent} 8.16.0

Review important information about the {fleet} and {agent} 8.16.0 release.

[discrete]
[[security-updates-8.16.0]]
=== Security updates

{fleet-server}::
* Update {fleet-server} Go version to 1.23.1. {fleet-server-pull}3924[#3924]

[discrete]
[[new-features-8.16.0]]
=== New features

The 8.16.0 release Added the following new and notable features.

{fleet}::
* Support content packages in UI. {kibana-pull}195831[#195831]

> * Advanced agent monitoring options UI for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]

Reviewer 2: ```suggestion
* Add advanced agent monitoring options for HTTP endpoint and diagnostics. {kibana-pull}193361[#193361]
```

* Create a task that periodically unenrolls inactive agents. Once an {agent} transitions to an `inactive` state and after a configurable timeout has expired, the agent will be unenrolled. {kibana-pull}189861[#189861]
* Add option to have Kafka dynamic topics in outputs. This allows the Kafka output to write to a topic which is dynamically set in an event field. {kibana-pull}192720[#192720]
* Add support for GeoIP processor databases in Ingest Pipelines. {kibana-pull}190830[#190830]
* Add agentless UX creation flow. {kibana-pull}189932[#189932]

> * Enable feature flag for reusable integration policies. This feature allows you to create integrations policies that can be shared with multiple {agent} policies, thereby reducing the number of integrations policies that you need to actively manage. {kibana-pull}187153[#187153]

Reviewer 2: ```suggestion
* Add support for reusable/shareable integration policies. This feature allows you to create integrations policies that can be shared with multiple {agent} policies, thereby reducing the number of integrations policies that you need to actively manage. {kibana-pull}187153[#187153]
```

> * Support integration-level outputs. This feature enables you to send integration data to a specific output, over-writing the output defined in the {agent} Policy. {kibana-pull}189125[#189125]

Reviewer 2: ```suggestion
* Add support for integration-level outputs. This feature enables you to send integration data to a specific output, overwriting the output defined in the {agent} Policy. {kibana-pull}189125[#189125]
```
